transport-msg.tex

```diff
@@ -617,15 +621,24 @@ \subsubsection{Driver Notifications}
         buffers and eventually notify the driver when they are used.
 \end{itemize}
 
-If a driver chooses not to use \msgref{EVENT_AVAIL}, or if the device does not
-support receiving it, the device \emph{MAY} rely on standard virtqueue polling
-mechanisms to detect virtqueue changes.
-
 \subsubsection{Device Notifications}
 \label{sec:Virtio Transport Options / Virtio Over Messages / Device Operation / Device Notifications}
 
-During normal operation, the device \emph{MAY} send notifications to the driver
-about configuration changes or the completion of buffers in virtqueues.
+During normal operation, the driver side bus \emph{MUST} send notifications to
+the driver about configuration changes, device status changes, or the
+completion of buffers in virtqueues using the messages described below.
+These notifications may be the result of:
+
+\begin{itemize}
+    \item The same messages received in band on the message channel from the
+        device side bus.
+    \item Manufactured by the driver side bus based on reception of an
+        out-of-band (OoB) notification from the device side.  Example OoB
+        notifications include things like specific MSIX or other IRQ signals.
+    \item Manufactured by the driver side bus periodically (polling).
+        This option should be reserved only for situation where nothing else
+        will work.
+  \end{itemize}
```

Comment:
This enumeration is not clear to me.

Reply:
The last manufactured by the driver periodically I do not get:
- do you mean the bus is generating "fake" events to force the drivers to poll ?
In FF-A we have kind of this case possible and the bus is polling periodically the device side for pending event so that it can generate events to the transport.
I do see how periodical fake events could work for virtqueues but not really for config for example.
Maybe something to discuss.

Reply:
Yes the bus invents messages.
"but not really for config for example."  
Yes it works for device status and config changes.  That is what EVENT_CONFIG0 was added for.  There are other was to do it like a bus message that is ASYNC_POLL

Reply:
I set this as a topic for the sprint.
I am happy to reword to make my intention clear or we can decide polling should be done differently